torch/distributions/categorical.py

```diff
@@ -93,11 +93,10 @@ def sample(self, sample_shape=torch.Size()):
     def log_prob(self, value):
         if self._validate_args:
             self._validate_sample(value)
-        value_shape = torch._C._infer_size(value.size(), self.batch_shape) if self.batch_shape else value.size()
-        param_shape = value_shape + (self._num_events,)
```

Comment:
is `self._num_events` always equal to `self.logits.size(-1)`?

Reply:
Yes